vnext/ReactUWP/Views/PopupViewManager.cpp

```diff
@@ -27,133 +27,206 @@ class PopupShadowNode : public ShadowNodeBase {
-  static void OnPopupClosed(IReactInstance &instance, int64_t tag, bool newValue);
-  void SetAnchorPosition(const winrt::Windows::UI::Xaml::Controls::Primitives::Popup &popup);
+
+  static void OnPopupClosed(IReactInstance &instance, int64_t tag);
   winrt::Windows::Foundation::Size GetAppWindowSize();
+  winrt::ContentControl GetControl();
+  void UpdateTabStops();
+  void UpdateLayout();
 
   bool IsWindowed() override {
     return true;
   }
 
  private:
-  std::unique_ptr<TouchEventHandler> m_touchEventHanadler;
+  bool m_autoFocus{true};
+  double m_verticalOffset{0};
+  double m_horizontalOffset{0};
+  std::unique_ptr<TouchEventHandler> m_touchEventHandler;
   std::unique_ptr<PreviewKeyboardEventHandlerOnRoot> m_previewKeyboardEventHandlerOnRoot;
 
   int64_t m_targetTag;
   winrt::Popup::Closed_revoker m_popupClosedRevoker{};
   winrt::Popup::Opened_revoker m_popupOpenedRevoker{};
+  winrt::FrameworkElement::SizeChanged_revoker m_popupSizeChangedRevoker{};
+  winrt::Window::SizeChanged_revoker m_windowSizeChangedRevoker{};
 };
 
 PopupShadowNode::~PopupShadowNode() {
-  m_touchEventHanadler->RemoveTouchHandlers();
+  m_touchEventHandler->RemoveTouchHandlers();
   m_previewKeyboardEventHandlerOnRoot->unhook();
 }
 
+winrt::ContentControl PopupShadowNode::GetControl() {
+  auto popup = GetView().as<winrt::Popup>();
+  auto control = popup.Child().as<winrt::ContentControl>();
+
+  return control;
+}
+
 void PopupShadowNode::createView() {
   Super::createView();
 
   auto popup = GetView().as<winrt::Popup>();
+  auto control = GetControl();
   auto wkinstance = GetViewManager()->GetReactInstance();
-  m_touchEventHanadler = std::make_unique<TouchEventHandler>(wkinstance);
+  m_touchEventHandler = std::make_unique<TouchEventHandler>(wkinstance);
   m_previewKeyboardEventHandlerOnRoot = std::make_unique<PreviewKeyboardEventHandlerOnRoot>(wkinstance);
 
   m_popupClosedRevoker = popup.Closed(winrt::auto_revoke, [=](auto &&, auto &&) {
     auto instance = wkinstance.lock();
     if (!m_updating && instance != nullptr)
-      OnPopupClosed(*instance, m_tag, false);
+      OnPopupClosed(*instance, m_tag);
   });
 
   m_popupOpenedRevoker = popup.Opened(winrt::auto_revoke, [=](auto &&, auto &&) {
     auto instance = wkinstance.lock();
     if (!m_updating && instance != nullptr) {
-      SetAnchorPosition(popup);
-
-      // When multiple flyouts/popups are overlapping, XAML's theme shadows
-      // don't render properly. As a workaround we enable a z-index
-      // translation based on an elevation derived from the count of open
-      // popups/flyouts. We apply this translation on open of the popup.
-      // (Translation is only supported on RS5+, eg. IUIElement9)
-      if (auto uiElement9 = GetView().try_as<winrt::IUIElement9>()) {
-        auto numOpenPopups = CountOpenPopups();
-        if (numOpenPopups > 0) {
-          winrt::Numerics::float3 translation{0, 0, (float)16 * numOpenPopups};
-          popup.Translation(translation);
```

Comment:
Can you describe the apparent blurriness that's caused by this code being removed?  By removing this code, won't the shadow problem return?

Reply:
Talked offline. The blurriness is caused by interaction with the perspective transform applied to the React root view. I've added the z-translation code back in for now and filed #4466 to track the underlying issue.